tests/integration/exec_test.py

```diff
@@ -88,6 +94,67 @@ def test_exec_command_streaming(self):
             res += chunk
         self.assertEqual(res, b'hello\nworld\n')
 
+    def test_exec_start_socket(self):
+        if not helpers.exec_driver_is_native():
+            pytest.skip('Exec driver not native')
+
+        container = self.client.create_container(BUSYBOX, 'cat',
+                                                 detach=True, stdin_open=True)
+        id = container['Id']
```

Comment:
Small thing: although we don't use Python's built-in `id` function anywhere in this block, it'd still be nicer not to shadow it for the sake of clarity (for one thing, it gets picked up by syntax highlighters, which results in misleading colouring). Maybe `container_id`?